rabbitmq/publisher_test.go

```diff
@@ -1,3 +1,152 @@
-package rabbitmq
+package rabbitmq_test
 
-// TODO: add unit tests
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/notifier-go/config"
+	"github.com/ElrondNetwork/notifier-go/data"
+	"github.com/ElrondNetwork/notifier-go/mocks"
+	"github.com/ElrondNetwork/notifier-go/rabbitmq"
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func createMockArgsRabbitMqPublisher() rabbitmq.ArgsRabbitMqPublisher {
+	return rabbitmq.ArgsRabbitMqPublisher{
+		Client: &mocks.RabbitClientStub{},
+		Config: config.RabbitMQConfig{
+			EventsExchange:          "allevents",
+			RevertEventsExchange:    "revert",
+			FinalizedEventsExchange: "finalized",
+		},
+	}
+}
+
+func TestRabbitMqPublisher(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil rabbitmq client", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsRabbitMqPublisher()
+		args.Client = nil
+
+		client, err := rabbitmq.NewRabbitMqPublisher(args)
+		require.True(t, check.IfNil(client))
+		require.Equal(t, rabbitmq.ErrNilRabbitMqClient, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgsRabbitMqPublisher()
+		client, err := rabbitmq.NewRabbitMqPublisher(args)
+		require.Nil(t, err)
+		require.NotNil(t, client)
+	})
+}
+
+func TestBroadcast(t *testing.T) {
+	t.Parallel()
+
+	done := make(chan struct{})
+
+	wasCalled := false
```

Comment:
since the writing of this var can be made asynchronous, you should protect this value with a mutex
Alternatively, you can use an uint32 var for this and the `atomic` package to manipulate the variable in a concurrent safe manner

Valid for the entire file

Reply:
thanks, changed to use atomic package